src/fsharp/pars.fsy

```diff
@@ -3306,6 +3306,8 @@ declExpr:
   | declExpr  INFIX_COMPARE_OP      declExpr { mkSynInfix (rhs parseState 2) $1 $2 $3 }
   | declExpr  DOLLAR                declExpr { mkSynInfix (rhs parseState 2) $1 "$" $3 }
   | declExpr  LESS                  declExpr { mkSynInfix (rhs parseState 2) $1 "<" $3 }
+  | declExpr  LESS                  recover { if not $3 then reportParseErrorAt (rhs parseState 2) (FSComp.SR.parsUnfinishedExpression("<"));  
+                                              exprFromParseError (mkSynInfix (rhs parseState 2) $1 "<" (arbExpr("declExprInfix",(rhs parseState 3).StartRange))) }
```

Comment:
This new parser recovery rule is needed when processing the incomplete text ``ID<``